openedx/stanford/djangoapps/sneakpeek/utils.py

```diff
@@ -0,0 +1,205 @@
+    anon_user.save()
+    profile = UserProfile(user=anon_user, nonregistered=True)
+    profile.save()
+    return anon_user
+
+
+def create_and_login_nonregistered_user(request):
+    """
+    Create a nonregistered user and log them in with a long session life
+    """
+    new_student = _create_nonregistered_user()
+    new_student.backend = settings.AUTHENTICATION_BACKENDS[0]
+    login(request, new_student)
+    # set session to very long to reduce number of nonreg users created
+    request.session.set_expiry(604800)
+
+
+def is_registered(user):
+    """
+    Handles django anonymous users.  SHOULD use this to test whether request.user has registered,
+    i.e. has a profile that says not nonregistered,
+    instead of directly accessing user.profile.nonregistered,
+    because if the user is AnonymousUser it won't have a profile.
+    """
+    registration = None
+    if hasattr(user, 'profile'):
+        registration = not user.profile.nonregistered
+    return registration
+
+
+def is_nonregistered(user):
+    """
+    Check if user is nonregistered (not Django Anonymous)
+    """
+    registration = is_registered(user) is False
+    return registration
+
+
+def deny_nonregistered_access(user, descriptor):
+    """
+    Check if nonregistered access should be denied for this user on this descriptor
+    """
+    should_deny = False
+    if is_nonregistered(user):
+        should_deny = descriptor.category not in NONREGISTERED_CATEGORY_WHITELIST
+    return should_deny
+
+
+def bind_within_enrollment_period(course):
+    """
+    Bind variables for the function
+    """
+    def within_enrollment_period():
+        """
+        Just a time boundary check, handles if start or stop were set to None
+        """
+        now = datetime.now(UTC)
+        start = course.enrollment_start
+        if start is not None:
+            start = start.replace(tzinfo=UTC)
+        end = course.enrollment_end
+        if end is not None:
+            end = end.replace(tzinfo=UTC)
+        return (start is None or now > start) and (end is None or now < end)
+    return within_enrollment_period
+
+
+def is_sneakpeek_allowed(user, course):
+    """
+    Check if nonregistered access is allow for a course
+
+    Sneakpeek is only allowed if:
+    1) within enrollment period
+    2) course specifies it's okay
+    3) request.user is not a registered user.
+    """
+    course_key = course.id
+    within_enrollment_period = bind_within_enrollment_period(course)
+    return (
+        within_enrollment_period()
+        and
+        CoursePreference.course_allows_nonregistered_access(course_key)
+        and
+        not user.is_registered()
+    )
+
+
+def is_regularly_registered(user, course):
+    """
+    Check if a user is enrolled in a course as a registered user
+    """
+    is_enrolled = registered_for_course(user, course)
+    return is_enrolled and user.is_registered()
+
+
+def registered_for_course(user, course):
```

Comment:
If you are changing function names then maybe 'enrolled_for_course'.

Reply:
@dcadams This call was refactored out entirely.